src/main.rs

```diff
@@ -95,17 +92,17 @@ async fn floodfill_pt(data: web::Data<AppState>, input: web::Json<UserInputJSON>
             count_original_nodes,
         ))
     }
-    
+
     // run for all in parallel
     let now = Instant::now();
     let parallel_res: Vec<(i32, u32, [i64; 32])> = model_parameters_each_start
         .par_iter()
-        .map(|input| floodfill(*input))
+        .map(|input| {
+            let (a, b, c, d, e, f, g, h, i) = *input;
```

Comment:
This is not a typical pattern, but the fact that floodfill takes so many arguments in the first place is a code smell. It looks like `read_files_serial` should maybe collect everything into one `struct`, and that should be passed around.

Relatedly, why are you using `Arc` now? It's so that you can push a bunch of stuff onto `model_parameters_each_start` and then later do `par_iter`. What if you "invert" where the logic happens, and just push the only thing that matters -- that `i` index -- onto the vec, and in parallel, do the rest?